engine/storage/snapshot/src/main/java/org/apache/cloudstack/storage/snapshot/SnapshotDataFactoryImpl.java

```diff
@@ -87,6 +87,9 @@ public SnapshotInfo getSnapshot(DataObject obj, DataStore store) {
     @Override
     public SnapshotInfo getSnapshot(long snapshotId, DataStoreRole role) {
```

Comment:
@DaanHoogland make sure 'null' returned here is checked, wherever this method is called.

Reply:
Note that at line 94/97 null is returned as well. It is called 20 times, not including tests. if the snapshot is null and an SnapshotInfo object is returned with a null snapshot field it will result in runtime exceptions if it is used. I agree that errors may occur in different placec, but not more errors will occur. I'll spend some time researching thos 20 callers.

Reply:
added two null checks. I think these are superfluent, but they wont hurt.